src/modules/accountsLink/services/telegram.accountsLink.service.ts

```diff
@@ -0,0 +1,444 @@
+          id: { $eq: id.trim() }
+        }
+      });
+
+    if (!existingTmpId) return null;
+    const ensuredEntity = await this.ensureTemporaryLinkingIdExpiration(
+      existingTmpId
+    );
+    if (ensuredEntity) return existingTmpId;
+    return null;
+  }
+
+  async getOrCreateTemporaryLinkingId(
+    parsedMessageWithDetails: SignedMessageWithDetails
+  ): Promise<TelegramTemporaryLinkingId> {
+    const existingLinkForSubstrateAccount =
+      await this.accountsLinkService.accountsLinkRepository.find({
+        where: {
+          substrateAccountId: {
+            $eq: parsedMessageWithDetails.address
+          },
+          notificationServiceName: NotificationServiceName.telegram,
+          active: true,
+          following: false
+        }
+      });
+
+    if (
+      existingLinkForSubstrateAccount &&
+      existingLinkForSubstrateAccount.length > 0
+    )
+      throw new ValidationError(
+        `Account ${parsedMessageWithDetails.address} already has linked Telegram account`
+      );
+
+    const existingTmpId = await this.getTemporaryLinkingIdBySubstrateAccount(
+      parsedMessageWithDetails.address
+    );
+    if (existingTmpId) return existingTmpId;
+
+    const newTmpIdEntity = new TelegramTemporaryLinkingId();
+    newTmpIdEntity.id = crypto.randomUUID();
+    newTmpIdEntity.substrateAccountId = parsedMessageWithDetails.address;
+    newTmpIdEntity.createdAt = new Date();
+
+    await this.telegramTemporaryLinkingIdRepository.save(newTmpIdEntity);
+    return newTmpIdEntity;
+  }
+
+  async getActiveLinkedTgAccountsBySubstrateAccount({
+    substrateAccount,
+    following
+  }: {
+    substrateAccount: string;
+    following: boolean;
+  }): Promise<LinkedTgAccountsToSubstrateAccountResponseType> {
+    const links = await this.accountsLinkService.accountsLinkRepository.find({
+      where: {
+        substrateAccountId: { $eq: substrateAccount },
+        notificationServiceName: { $eq: NotificationServiceName.telegram },
+        active: { $eq: true },
+        following: { $eq: following }
+      }
+    });
+
+    const telegramAccountsMap = new Map<string, TelegramAccount>(
+      (
+        await this.telegramAccountRepository.find({
+          where: {
+            accountId: {
+              $in: links.map((link) => link.notificationServiceAccountId)
+            }
+          }
+        })
+      ).map((tgAcc) => [tgAcc.accountId, tgAcc])
+    );
+
+    return {
+      telegramAccounts: links.map((link) => {
+        const tgAccount = telegramAccountsMap.get(
+          link.notificationServiceAccountId
+        );
+        return {
+          accountId: tgAccount ? tgAccount.accountId : '',
+          userName: tgAccount ? tgAccount.userName : '',
+          firstName: tgAccount ? tgAccount.firstName : '',
+          lastName: tgAccount ? tgAccount.lastName : '',
+          phoneNumber: tgAccount ? tgAccount.phoneNumber : ''
+        };
+      })
+    };
+  }
+
+  async ensureTelegramAccount({
```

Comment:
Why it is ensure, if you are creating new one

Reply:
because it's more `getOrCreateTelegramAccount` - it will return existing or will create new one.